### PR TITLE
Improve social previews metadata in React

### PR DESCRIPTION
**Pull request by rpalakkal**
addresses #1525

### CODE REVIEW

File: src/components/common/Layout.tsx [hunk 1]
```diff
@@ -9,6 +9,24 @@ import Header from "./Header";
  * place in <head> through React Helmet.
  */
 export default class Layout extends React.Component<LayoutProps> {
+  title: string;
+  description: string;
+
+  /**
+   * Initialize the component and its state.
```
Comment (Ceshion) [inline src/components/common/Layout.tsx:16]: nitpicking but- this one is stateless 😄 (in the sense of, does not have mutable state and only receives data from its parent) so we just "Initialize the component"

File: src/components/email/Email.tsx [hunk 2]
```diff
@@ -93,7 +93,7 @@ export default class Email extends React.Component<
    */
   render(): React.ReactNode {
     return (
-      <Layout {...this.siteConfig}>
+      <Layout {...this.siteConfig} {...this.emailData}>
```
Comment (Ceshion) [inline src/components/email/Email.tsx:96]: we aren't using the entirety of emailData in `<Layout>`, so maybe 
`<Layout {...this.siteConfig} city={this.emailData.city} state={this.emailData.state}>`? if you're feeling ambitious about re-decoupling `<Layout>` from `Email`s, maybe even mutate `siteConfig` in this class and just pass the updated `title` and `meta` as properties ❤️

File: src/types/PropTypes.ts [hunk 3]
```diff
@@ -26,8 +26,8 @@ export interface EmailProps {
 
 // {Static query result properties}
 export type LayoutProps = Pick<
-  SiteConfig,
-  "title" | "meta" | "faviconUrl" | "logoUrl"
+  SiteConfig & EmailMetadata,
+  "title" | "meta" | "faviconUrl" | "logoUrl" | "name" | "city" | "state"
```
Comment (Ceshion) [inline src/types/PropTypes.ts:30]: are we using `EmailMetadata.name`?

File: src/components/common/Header.tsx [hunk 4]
```diff
@@ -17,9 +17,7 @@ class _Header extends React.Component<HeaderProps> {
     return (
       <header className="header">
         <Link to="/">
-          <h1 aria-label="Defund Twelve .org" className="pagetitle">
```
Comment (rpalakkal) [inline src/components/common/Header.tsx:20]: className="pagetitle" not being used I believe

**Pull request merged**